tests/test_frompyobject.rs

```diff
@@ -299,3 +299,34 @@ fn test_err_rename() {
         "TypeError: 'dict' object cannot be converted to 'Union[str, uint, int]'"
     );
 }
+
+fn to_string(obj: &PyAny) -> PyResult<String> {
+    Ok(obj.str()?.to_str()?.to_string())
+}
+
+#[derive(Debug, FromPyObject)]
+pub struct Zap {
+    #[pyo3(item)]
+    name: String,
+
+    #[pyo3(from_py_with = "to_string", item("my_object"))]
+    some_object: String,
+}
+
+#[test]
```

Comment:
Final thought; could we please add tests for tuple structs and enum variants? Just to make sure that all cases are covered and remain covered.

Reply:
Sure, thank you for the reminder!

Reply:
I've finally got to this and I have a question. At the moment we provide `from_py_with` for struct fields only.
Do you have in mind that `from_py_with` should be supported for tuple structs and enums as well?

Reply:
Hmm, that's a good point, I didn't really specify those on #1239 when I proposed the idea.

I think for tuple structs with `#[derive(FromPyObject)]` it's quite natural to want to be able to annotate fields in the same way. Maybe let's skip on deciding what to do with enum variants for now. That'll be easy to add later anyway.

If tuple structs are going to be a lot of extra work, perhaps let's merge this PR for now and add an issue to remember to support them later?

Reply:
Thanks for clearing this out! I'll try supporting tuple structs, maybe it won't be that hard.
In any case, I think it's worth adding a compiler error (warning?) when `from_py_with` is used on types that are not supported